internal/cmd/operator-sdk/alpha/config3alphato3/convert_config_3-alpha_to_3.go

```diff
@@ -0,0 +1,214 @@
+// Copyright 2021 The Operator-SDK Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package config3alphato3
+
+import (
+	"bufio"
+	"bytes"
+	"errors"
+	"io/ioutil"
+	"path"
+	"regexp"
+	"strings"
+	"text/template"
+
+	"golang.org/x/mod/modfile"
+	"sigs.k8s.io/kubebuilder/v3/pkg/model/resource"
+	"sigs.k8s.io/yaml"
+)
+
+var (
+	v3alpha   = "3-alpha"
+	versionRe = regexp.MustCompile(`version:[ ]*(?:")?3-alpha(?:")?`)
+)
+
+func convertConfig3AlphaTo3(cfgBytes []byte) (_ []byte, err error) {
+	cfgObj := make(map[string]interface{}, 5)
+	if err := yaml.Unmarshal(cfgBytes, &cfgObj); err != nil {
+		return nil, err
+	}
+
+	if cfgObj["version"].(string) != v3alpha {
+		return cfgBytes, nil
+	}
+
+	cfgBytes = versionRe.ReplaceAll(cfgBytes, []byte(`version: "3"`))
+
+	var modulePath string
+	layout := cfgObj["layout"].(string)
+	isGo := strings.HasPrefix(layout, "go.kubebuilder.io/")
+	if isGo {
+		if modulePath, err = getModulePath(); err != nil {
+			return nil, err
+		}
+	}
+
+	multigroupObj, hasMultigroup := cfgObj["multigroup"]
+	isMultigroup := hasMultigroup && multigroupObj.(bool)
+
+	if obj, hasRes := cfgObj["resources"]; hasRes {
+		domain := cfgObj["domain"].(string)
+		resObjs := obj.([]interface{})
+		resources := make([]resource.Resource, len(resObjs))
+
+		for i, resObj := range resObjs {
+			resources[i].Domain = domain
+
+			res := resObj.(map[string]interface{})
+			if groupObj, ok := res["group"]; ok {
+				resources[i].Group = groupObj.(string)
+			}
+			if versionObj, ok := res["version"]; ok {
+				resources[i].Version = versionObj.(string)
+			}
+			if kindObj, ok := res["kind"]; ok {
+				resources[i].Kind = kindObj.(string)
+			}
+
+			if k8sDomain, found := coreGroups[resources[i].Group]; found {
```

Comment:
There are some cases where a group like "apps" can be used without being a core-resource, e.g. the CronJob type in the tutorial.

Reply:
Fixed and added a test case.